src/test_fixture/fabric.rs

```diff
@@ -0,0 +1,197 @@
+use std::collections::hash_map::Entry;
```

Comment:
I didn't look at this closely.  There are a few things here that worry me in terms of complexity.  See earlier comments on some ideas that might help.  Later.